File: cpp/darts/darts.cpp
#include "darts.h"

namespace darts
{
    int score(float x, float y)
    {
        if (x * x + y * y <= 1)
        {
            return 10;
        }
        if (x * x + y * y <= 5 * 5)
        {
            return 5;
        }
        if (x * x + y * y <= 10 * 10)
        {
            return 1;
        }
        return 0;
    }
} // namespace darts

File: go/darts/darts.go
package darts

import (
	"math"
)

const (
	outerRadius  = 10
	middleRadius = 5
	innerRadius  = 1
)

func Score(x, y float64) int {
	distanceToCenter := math.Sqrt(x*x + y*y)
	switch {
	case distanceToCenter > outerRadius:
		return 0
	case distanceToCenter > middleRadius:
		return 1
	case distanceToCenter > innerRadius:
		return 5
	default:
		return 10
	}
}
